apps/ensrainbow/src/lib/database.ts

```diff
@@ -11,18 +11,21 @@ export const LABELHASH_COUNT_KEY = new Uint8Array([0xff, 0xff, 0xff, 0xff]) as B
 export const INGESTION_IN_PROGRESS_KEY = new Uint8Array([0xff, 0xff, 0xff, 0xfe]) as ByteArray;
 
 /**
- * Checks if there's an incomplete ingestion and exits with a helpful error message if one is found
+ * Checks if there's an incomplete ingestion and throws an error if one is found
  * @param db The ENSRainbow database instance
+ * @throws Error if an incomplete ingestion is detected
  */
-export async function exitIfIncompleteIngestion(db: ENSRainbowDB): Promise<void> {
+export async function checkIngestionState(db: ENSRainbowDB): Promise<void> {
```

Comment:
```suggestion
export async function ensureIngestionNotIncomplete(db: ENSRainbowDB): Promise<void> {
```

Suggesting this because the function doesn't just check, it ensures. And it specifically ensures that an ingestion is not incomplete.